### PR TITLE
Improved tests.

### PR DESCRIPTION
**Pull request by radekmie**
In this pull request, I improved the test coverage and got rid of the warning in Material UI theme.

**Update 2023-03-09:** I moved some tests to RTL. That required an updated of `@testing-library/*` dependencies and TypeScript 4.4.

### CODE REVIEW

File: packages/uniforms-material/src/ErrorsField.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import { useTheme, PropTypes } from '@material-ui/core';
+import type { PropTypes } from '@material-ui/core';
 import FormControl from '@material-ui/core/FormControl';
 import FormHelperText, {
   FormHelperTextProps,
 } from '@material-ui/core/FormHelperText';
+import useTheme from '@material-ui/core/styles/useTheme';
```
Comment (Monteth) [inline packages/uniforms-material/src/ErrorsField.tsx:6]: Is it faster?

**Review by ErnestTeluk — APPROVED** [review]

File: packages/uniforms/__tests__/AutoForm.tsx [hunk 2]
```diff
@@ -144,51 +152,90 @@ describe('<AutoForm />', () => {
...
+      const input = screen.getByLabelText('A');
+      fireEvent.change(input, { target: { value: 'a' } });
+
+      expect(contextSpy).toHaveBeenLastCalledWith(
+        expect.objectContaining({ model: { a: 'a' } }),
       );
-      const { rerender } = render(<Component />, schemaDefinition);
 
-      rerender(<Component />);
+      contextSpy.mock.calls.slice(-1)[0][0]?.formRef.reset();
```
Comment (Monteth) [inline packages/uniforms/__tests__/AutoForm.tsx:175]: Wow, don't we want to have a reusable function to reset the contextSpy? In this form it looks terrifying

File: packages/uniforms/__tests__/AutoForm.tsx [hunk 3]
```diff
@@ -65,21 +67,21 @@ describe('<AutoForm />', () => {
     });
 
     it('calls `onChangeModel`', () => {
+      const schema = new SimpleSchema({ a: { type: String, optional: true } });
+      const bridge = new SimpleSchema2Bridge(schema);
```
Comment (Monteth) [inline packages/uniforms/__tests__/AutoForm.tsx:71]: Don't we wanna switch to zod also here?

File: packages/uniforms-antd/__tests__/index.ts [hunk 4]
```diff
@@ -28,3 +29,14 @@ it('exports everything', () => {
...
+
+describe('@RTL', () => {
+  suites.testAutoField(antd.AutoField);
+  // FIXME: AntD `DatePicker` is far from the HTML one.
+  // suites.testDateField(antd.DateField);
+  suites.testListDelField(antd.ListDelField);
+  suites.testListField(antd.ListField, {
+    getListAddField: screen => screen.getByRole('img', { name: 'plus-square' }),
+  });
+  suites.testTextField(antd.TextField);
```
Comment (Monteth) [inline packages/uniforms-antd/__tests__/index.ts:41]: I'm not sure if I like moving suite tests to the index files while we still have some tests in a separate file.

File: packages/uniforms-material/src/ErrorsField.tsx (hunk 1 above)
Comment (radekmie) [reply packages/uniforms-material/src/ErrorsField.tsx:6]: Nope, but it got rid of [this warning](https://github.com/vazco/uniforms/actions/runs/4384694127/jobs/7676475057#step:8:16).

File: packages/uniforms/__tests__/AutoForm.tsx (hunk 2 above)
Comment (radekmie) [reply packages/uniforms/__tests__/AutoForm.tsx:175]: Now that I look at it... 😅 I'll improve it in the next PRs.

File: packages/uniforms/__tests__/AutoForm.tsx (hunk 3 above)
Comment (radekmie) [reply packages/uniforms/__tests__/AutoForm.tsx:71]: Sure, but maybe not in this PR - I'd rather iterate faster on these.

File: packages/uniforms-antd/__tests__/index.ts (hunk 4 above)
Comment (radekmie) [reply packages/uniforms-antd/__tests__/index.ts:41]: Got it. I wanted to do it this way just to force us into centralizing this "config" and improving on it later... Let's discuss it and if needed, I'll move it back.

File: packages/uniforms-material/src/ListDelField.tsx [hunk 5]
```diff
@@ -30,19 +30,16 @@ function ListDel({
     { absoluteName: true },
   )[0];
 
-  const limitNotReached =
-    !disabled && !(parent.minCount! >= parent.value!.length);
+  disabled ||= readOnly || parent.minCount! >= parent.value!.length;
```
Comment (kestarumper) [inline packages/uniforms-material/src/ListDelField.tsx:33]: I know it's not an object, but is it a good practice to override the `disabled` value?

File: packages/uniforms/__suites__/ListDelField.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  test('<ListDelField> - correctly reacts on enter', async () => {
+    const onChange = jest.fn();
+    renderWithZod({
+      element: <ListDelField name="x.1" data-testid="x" />,
+      model: { x: ['a', 'b', 'c'] },
+      onChange,
+      schema: z.object({ x: z.array(z.string()) }),
+    });
+    await userEvent.type(screen.getByTestId('x'), '{Enter}');
```
Comment (kestarumper) [inline packages/uniforms/__suites__/ListDelField.tsx:31]: Not sure if this is a typo or not, [but in the documentation](https://testing-library.com/docs/ecosystem-user-event/#special-characters), it's lowercase `{enter}`. No idea if this has any impact.

**Pull request merged**